cmake/Ekat.cmake

```diff
@@ -170,45 +138,62 @@ macro (setVars PREFIX SET_DEFAULTS)
     set (EKAT_DISABLE_TPL_WARNINGS OFF CACHE BOOL "")
   endif()
 
-  # Packs variables
-  if (DEFINED ${PREFIX}_PACK_SIZE)
-    set (EKAT_PACK_SIZE ${${PREFIX}_PACK_SIZE} CACHE STRING "")
+  if (DEFINED ${PREFIX}_ENABLE_TESTS)
+    set (EKAT_ENABLE_TESTS ${${PREFIX}_ENABLE_TESTS} CACHE BOOL "")
   elseif (SET_DEFAULTS)
-    if (EKAT_CUDA_BUILD)
-      set (EKAT_PACK_SIZE 1 CACHE STRING "")
-    else ()
-      set (EKAT_PACK_SIZE 16 CACHE STRING "")
-    endif()
+    set (EKAT_ENABLE_TESTS ON CACHE BOOL "")
   endif()
 
-  if (DEFINED ${PREFIX}_SMALL_PACK_SIZE)
-    set (EKAT_SMALL_PACK_SIZE ${${PREFIX}_SMALL_PACK_SIZE} CACHE STRING "")
+  ### Needed only if EKAT_ENABLE_TESTS=ON ###
+
+  if (DEFINED ${PREFIX}_TEST_MAX_THREADS)
+    set (EKAT_TEST_MAX_THREADS ${${PREFIX}_TEST_MAX_THREADS} CACHE STRING "")
   elseif (SET_DEFAULTS)
-    set (EKAT_SMALL_PACK_SIZE ${EKAT_PACK_SIZE} CACHE STRING "")
+    set (EKAT_TEST_MAX_THREADS 1 CACHE STRING "")
   endif()
 
-  if (DEFINED ${PREFIX}_PACK_CHECK_BOUNDS)
-    set (EKAT_PACK_CHECK_BOUNDS ${${PREFIX}_PACK_CHECK_BOUNDS} CACHE BOOL "")
+  if (DEFINED ${PREFIX}_TEST_THREAD_INC)
+    set (EKAT_TEST_THREAD_INC ${${PREFIX}_TEST_THREAD_INC} CACHE STRING "")
+  elseif (SET_DEFAULTS)
+    set (EKAT_TEST_THREAD_INC 1 CACHE STRING "")
   endif()
 
-  if (DEFINED ${PREFIX}_POSSIBLY_NO_PACK)
-    set (EKAT_POSSIBLY_NO_PACK ${${PREFIX}_POSSIBLY_NO_PACK} CACHE BOOL "")
+  if (DEFINED ${PREFIX}_TEST_STRICT_FP)
+    set (EKAT_TEST_STRICT_FP ${${PREFIX}_TEST_STRICT_FP} CACHE STRING "")
+  elseif (setVars_DEBUG_BUILD AND SET_DEFAULTS)
+    set (EKAT_TEST_STRICT_FP ON CACHE BOOL "")
+  endif()
+
+  if (DEFINED ${PREFIX}_TEST_PACK_SIZE)
+    set (EKAT_TEST_PACK_SIZE ${${PREFIX}_TEST_PACK_SIZE} CACHE STRING "")
+  elseif (SET_DEFAULTS)
+    if (EKAT_CUDA_BUILD)
+      set (EKAT_TEST_PACK_SIZE 1 CACHE STRING "")
+    else ()
+      set (EKAT_TEST_PACK_SIZE 16 CACHE STRING "")
+    endif()
   endif()
 
-  if (DEFINED ${PREFIX}_POSSIBLY_NO_PACK_SIZE)
-    set (EKAT_POSSIBLY_NO_PACK_SIZE ${${PREFIX}_POSSIBLY_NO_PACK_SIZE} CACHE STRING "")
+  if (DEFINED ${PREFIX}_TEST_SMALL_PACK_SIZE)
```

Comment:
Just curious: what's a "small pack size"? Is this for GPU stuff that doesn't use vectorization, as suggested by POSSIBLY_NO_PACK_SIZE?

Reply:
So, this is linked to the fact that, IIRC, in the pack seminal work, Jim and Andrew found it is actually ok (or maybe even a tad better) to use a pack size that is larger than the arch's vector unit (of course, it must be a multiple of that). However, some loops might benefit from having smaller packs (like a single vector unit). Therefore, scream introduced a "small" pack size (like a single VU), as opposed to the "normal" pack size (a multiple of a single VU).

Reply:
POSSIBLY_NO_PACK_SIZE is honestly a bit confusing. It was introduced for the LinInterp mini app, where, on SKX, Jim observed that a pack size of 1 performs better than a pack size of 8 (or more). This is somewhat counterintuitive, especially considering pack size 8 is better than 1 on other arch (like KNL).

So that pack size is for algorithms such as lin interp which on SKX seem to prefer no pack, but on other archs still want a pack size larger than 1.

Reply:
Imho, scream should NOT use a pack size at the SCREAM level, but rather have each individual parametrization use their own pack size. That's one of the reasons I purged the pack size from the configuration of the library (the config options here above are only used in the testing of ekat).